cfme/infrastructure/pxe.py

```diff
@@ -156,15 +254,32 @@ def create(self, cancel=False, refresh=True, refresh_timeout=120):
             refresh (boolean): Whether to run the refresh operation on the PXE server after
                 the add has been completed.
         """
-        navigate_to(self, 'Add')
-        fill(pxe_properties_form, self._form_mapping(True, **self.__dict__))
-        self._submit(cancel, form_buttons.add)
-        if not cancel:
-            flash.assert_message_match('PXE Server "{}" was added'.format(self.name))
+        view = navigate_to(self, 'Add')
+        view.fill({'name': self.name,
```

Comment:
@mfalesni, we have discussed this the other day. Yes we are going to make update method return changed. that method should click cancel if there were no any changes. tests should check returned value.